conans/test/unittests/test_client/test_graph/test_range_resolver.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+from conans.client.graph.range_resolver import _parse_versionexpr
+from conans.test.utils.tools import TestBufferConanOutput
+
+
+class ParseVersionExpr(unittest.TestCase):
+    def test_backwards_compatibility(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, <=3.2", output), ("2.3 <=3.2", True, False))
+
+    def test_only_loose(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3 ,3.2, loose=True", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2, loose=False", output), ("2.3 3.2", False, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2, loose  = False", output), ("2.3 3.2", False, False))
+        self.assertEqual(_parse_versionexpr("2.3 3.2,  loose  = True", output), ("2.3 3.2", True, False))
+
+    def test_only_prerelease(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=False", output), ("2.3 3.2", True, False))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=True", output), ("2.3 3.2", True, True))
+
+    def test_both(self):
+        output = TestBufferConanOutput()
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, loose=False, include_prerelease=True", output),
+                         ("2.3 3.2", False, True))
+        self.assertEqual(_parse_versionexpr("2.3, 3.2, include_prerelease=True, loose=False", output),
+                         ("2.3 3.2", False, True))
+
+    def test_invalid(self):
```

Comment:
I think that invalid inputs should raise a ConanException; otherwise, what should Conan do with them? What is the output of `max_satisfying` if provided with an invalid version range?

Reply:
I think that your comment should be in this _thread_
> I tested it directly and it delivers no results back and unluckily no exception. This should be put into a new issue to follow up later (when something changed in the lib?).

If `max_satisfying` returns nothing, the next line will raise a `KeyError` (`return candidates.get(result)`), so I think that it is worth handling the error or raise a meaninful exception (`ConanException` telling that the version range has an invalid format).

Reply:
I added now an `ConanException`, when the range is invalid. Unluckily this is not working for all of your invalid ranges, because the semver library is accepted most of them, when `loose=True`. But this would be more a feature/bug in the semver library.

Reply:
Your exception is being raised if `semver.Range` raises. We cannot rely on `semver` satisfying our documented behavior; we have to check on our side and pass only what we consider a valid string to `semver`. We have to be at least as restrictive as `semver` is (and will be).

Reply:
I am not able to check, if the range is valid on the Conan side, because I don't know exact the semver syntax. And as long we have to live with the "comma", we are not able to check this on our side. Please put this into another issue, so that this issue comes through. It's an really important feature for "CI".

Reply:
We have to check the string on the Conan side, it has nothing to do with `semver` library. In the future we may move to another _semver_ library and we will have to maintain backwards compatibility, so the valid/invalid checks should be implemented on our side (we are only delegating the `max_satisfying` calculation to the library, but we could do it ourselves too).

I'm trying to get a regex to validate this string. I have this one, but still fails with some use cases.

```
[a-zA-Z0-9_,<>=.\s~]{2,}?(\s*,\s*(loose|include_prerelease)\s*=\s*(True|False)){1,2}
```

Reply:
I don't know how to do it with only one regex... but take a look at this proposal (it is built on top of your PR, feel free to merge from my branch): https://github.com/jgsogo/conan/commit/9d5192368021ce4fd6561122813b4fe752b0d1d4

Reply:
Look's good for me. I merged it into my branch.